scapy/layers/dot15d4.py

```diff
@@ -303,12 +331,137 @@ class Dot15d4Beacon(Packet):
+                        tmp_val = self.sec_payload[i]
+                    self.gts_dir_reserved = (tmp_val >> 7) & 0b1
+                    self.gts_dir_mask = tmp_val & 0b1111111
+                    i += 1
+                    # TODO: Update the GTS List field
+            if len(self.sec_payload) > i:
+                if isinstance(self.sec_payload[i], str):
+                    tmp_val = struct.unpack(">B", self.sec_payload[i])[0]
+                else:
+                    tmp_val = self.sec_payload[i]
+                self.pa_reserved_1 = (tmp_val >> 7) & 0b1
+                self.pa_num_long = (tmp_val >> 4) & 0b111
+                self.pa_reserved_2 = (tmp_val >> 3) & 0b1
+                self.pa_num_short = tmp_val & 0b111
+                i += 1
+            for _ in range(self.pa_num_short):
+                if len(self.sec_payload) > i + 1:
+                    if isinstance(self.sec_payload[i], str):
+                        tmp_val = (
+                            struct.unpack("<H", self.sec_payload[i:i + 2])[0]
+                        )
+                    else:
+                        tmp_val = (
+                            self.sec_payload[i] +
+                            (self.sec_payload[i + 1] << 8)
+                        )
+                    self.pa_short_addresses.append(tmp_val)
+                    i += 2
+            for _ in range(self.pa_num_long):
+                if len(self.sec_payload) > i + 7:
+                    if isinstance(self.sec_payload[i], str):
+                        tmp_val = (
+                            struct.unpack("<Q", self.sec_payload[i:i + 8])[0]
+                        )
+                    else:
+                        tmp_val = (
+                            self.sec_payload[i] +
+                            (self.sec_payload[i + 1] << 8) +
+                            (self.sec_payload[i + 2] << 16) +
+                            (self.sec_payload[i + 3] << 24) +
+                            (self.sec_payload[i + 4] << 32) +
+                            (self.sec_payload[i + 5] << 40) +
+                            (self.sec_payload[i + 6] << 48) +
+                            (self.sec_payload[i + 7] << 56)
+                        )
+                    self.pa_long_addresses.append(tmp_val)
+                    i += 8
+            self.sec_payload = self.sec_payload[i:]
+        return s
+
     def mysummary(self):
         return self.sprintf("802.15.4 Beacon ( %Dot15d4Beacon.src_panid%:%Dot15d4Beacon.src_addr% ) assocPermit(%Dot15d4Beacon.sf_assocpermit%) panCoord(%Dot15d4Beacon.sf_pancoord%)")  # noqa: E501
 
+    def guess_payload_class(self, payload):
+        from scapy.layers.sixlowpan import ThreadBeacon
+        from scapy.layers.zigbee import ZigBeeBeacon
+        if len(payload) > 0:
+            if (
+                (isinstance(payload[0], int) and payload[0] == 0x03) or
+                (isinstance(payload[0], bytes) and payload[0] == b'\x03')
```

Comment:
You can use `orb(payload[0]) == 0x03` to work on all Python versions (`from scapy.compat import orb`)